docs/release.md

```diff
@@ -125,6 +125,18 @@ When considering making a bugfix release on the `v0.N.x` release cycle, the bug
 
 The OpenTelemetry Collector maintainers will ultimately have the responsibility to assess if a given bug fulfills all the necessary criteria and may grant exceptions in a case-by-case basis.
 
+### Bugfix release procedure
+
+The following documents the procedure to release a bugfix
+
+1. Create a pull request against the `release/<release-series>` (e.g. `release/v0.45.x`) branch to apply the fix.
+2. Create a pull request to update version number against the `release/<release-series>` branch.
+3. Once those changes have been merged, create a pull request to the `main` branch from the `release/<release-series>` branch.
+4. Enable the **Merge pull request** setting in the repository's **Settings** tab.
+5. **IMPORTANT**: The pull request to bring the changes from the release branch *MUST* be merged using the **Merge pull request** method, and *NOT* squashed using “**Squash and merge**”. This is important as it allows us to ensure the commit SHA from the release branch is also on the main branch. **Not following this step will cause much go dependency sadness.**
```

Comment:
One small caveat that happens, after you "merge pull request" the branch will get deleted :) Just add a note to "restore" the branch.